src/components/CreditCard.test.js

```diff
@@ -0,0 +1,119 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+
+import CreditCard from './CreditCard';
+
+const providersData = [
+    { name: 'American Express', cardNumber: '378282246310005' },
+    { name: 'Visa', cardNumber: '4701322211111234' },
+    { name: 'MasterCard', cardNumber: '5555555555554444' },
+    { name: 'Diners Club', cardNumber: '3000 0000 0000 04' },
+    { name: 'JCB', cardNumber: '3088 0000 0000 0009' },
+];
+
+function setup() {
+    render(<CreditCard />);
+}
+
+describe('CreditCard', () => {
+    it('should render credit card input', () => {
+        setup();
+
+        const inputEl = screen.getByRole('textbox', { name: /card number/i });
+        expect(inputEl).toBeInTheDocument();
+    });
+
+    it('should render submit button', () => {
+        setup();
+
+        const button = screen.getByRole('button', { name: /send/i });
+        expect(button).toBeInTheDocument();
+    });
+
+    it('should display an error if card number is not valid', async () => {
+        const invalidCardNumber = '123321';
+        setup();
+
+        // dostaję tutaj warning, żeby użyć act, bo jest zmieniany state
+        // a jak używam act, to eslint mówi, żeby w testach jednoskowych nie używać act
+        const inputEl = screen.getByRole('textbox', { name: /card number/i });
+        userEvent.type(inputEl, invalidCardNumber);
+
+        const button = screen.getByRole('button', { name: /send/i });
+        userEvent.click(button);
+
+        const error = await screen.findByText('card number invalid');
+        expect(error).toBeInTheDocument();
+
+        // jeżeli dobrze rozumiem, to w tej sytuacji, nie muszę używać waitFor, bo mamy
+        // tylko przeładowanie stanu, nic kompleksowego
+        // jak api call i przeładowanie stanu? (pewnie to również bez waitFor można zrobić)
+        // czy może w waitFor bardziej chodzi o rzeczy jak setInterval?
```

Comment:
W zasadzie działa to podobnie (patrz komentarz wyżej). 

Próbowałeś zrobić to w podobny sposób jak w tym artykule?
https://kentcdodds.com/blog/fix-the-not-wrapped-in-act-warning